src/main/java/net/fabricmc/loader/impl/discovery/ResultAnalyzer.java

```diff
@@ -0,0 +1,287 @@
+							addErrorToList(mod, dep, depMod, pw);
+						}
+
+						break;
+					case CONFLICTS:
+						depMod = selectedMods.get(dep.getModId());
+
+						if (depMod != null && dep.matches(depMod.getVersion())) {
+							addErrorToList(mod, dep, depMod, pw);
+						}
+
+						break;
+					default:
+						// ignore
+					}
+				}
+			}
+		}
+
+		if (sw.getBuffer().length() == 0) {
+			return null;
+		} else {
+			return sw.toString();
+		}
+	}
+
+	private static void addErrorToList(ModCandidate mod, ModDependency dep, ModCandidate match, PrintWriter pw) {
+		addErrorToList(mod, dep, match != null ? Collections.singletonList(match) : Collections.emptyList(), pw);
+	}
+
+	private static void addErrorToList(ModCandidate mod, ModDependency dep, List<ModCandidate> matches, PrintWriter pw) {
+		if (matches == null) matches = Collections.emptyList();
+
+		matches = matches.stream().filter(m -> dep.matches(m.getVersion())).collect(Collectors.toList());
+
+		String errorType;
+
+		switch (dep.getKind()) {
+		case DEPENDS: errorType = "requires"; break;
+		case RECOMMENDS: errorType = "recommends"; break;
+		case BREAKS: errorType = "is incompatible with"; break;
+		case CONFLICTS: errorType = "conflicts with"; break;
+		default: throw new IllegalStateException("unknown dep kind: "+dep.getKind());
+		}
+
+		pw.printf("\n - Mod %s %s %s of mod %s, ",
+				getCandidateName(mod), errorType, getDependencyVersionRequirements(dep),
+				(matches.isEmpty() ? dep.getModId() : getCandidateName(matches.get(0))));
+
+		if (matches.isEmpty()) {
+			appendMissingDependencyError(dep, pw);
+		} else if (dep.getKind().isPositive()) {
+			appendUnsatisfiedDependencyError(dep, matches, pw);
+		} else if (dep.getKind() == Kind.CONFLICTS) {
+			appendConflictError(mod, matches, pw);
+		} else {
+			appendBreakingError(mod, matches, pw);
+		}
+
+		if (matches != null) {
+			appendJiJInfo(matches, pw);
+		}
+	}
+
+	private static void appendMissingDependencyError(ModDependency dependency, PrintWriter pw) {
+		pw.printf("which is missing!\n\t - You must install %s of %s.",
+				getDependencyVersionRequirements(dependency), dependency.getModId());
+	}
+
+	private static void appendUnsatisfiedDependencyError(ModDependency dependency, List<ModCandidate> matches, PrintWriter pw) {
+		pw.printf("but a non-matching version is present: %s!\n\t - You must install %s of %s.",
+				getCandidateFriendlyVersions(matches), getDependencyVersionRequirements(dependency), getCandidateName(matches.get(0)));
+	}
+
+	private static void appendConflictError(ModCandidate candidate, List<ModCandidate> matches, PrintWriter pw) {
+		final String depCandidateVer = getCandidateFriendlyVersions(matches);
+
+		pw.printf("but a matching version is present: %s!\n"
+				+ "\t - While this won't prevent you from starting the game,  the developer(s) of %s have found that "
+				+ "version %s of %s conflicts with their mod.\n"
+				+ "\t - It is heavily recommended to remove one of the mods.",
```

Comment:
Is it heavily recommended or just recommended? If the conflict was that fatal surely it would just be breaks?

Reply:
These are still the old texts, just moved to a different file, should probably be reworded some when redoing the explaining frontend.

Reply:
fixed a few of these now where it's particularly obvious